### PR TITLE
UCT/MM/SELF/ATOMIC: added bitwise ops

### PR DESCRIPTION
**Pull request by hoopoepg**

### CODE REVIEW

File: src/ucs/arch/atomic.h [hunk 1]
```diff
@@ -19,6 +19,42 @@
 #  error "Unsupported architecture"
 #endif
 
+#define UCS_DEFINE_ATOMIC_AND(wordsize, suffix) \
```
[the remark this answers is not in the document]
Comment (hoopoepg) [reply src/ucs/arch/atomic.h]: here macro arg is used as literal, but ok, updated
suffix is used for arch-specific macro expansion, I just unified interface (for future implementation)

File: src/uct/sm/base/sm_ep.c [hunk 2]
```diff
@@ -54,40 +54,168 @@ ucs_status_t uct_sm_ep_get_bcopy(uct_ep_h tl_ep, uct_unpack_callback_t unpack_cb
...
+    default:
+        ucs_assertv(0, "incorrect opcode: %d", opcode);
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    UCT_TL_EP_STAT_ATOMIC(ucs_derived_of(ep, uct_base_ep_t));
+    return UCS_OK;
+}
+
+ucs_status_t uct_sm_ep_atomic64_post(uct_ep_h ep, unsigned opcode, uint64_t value,
```
[the remark this answers is not in the document]
Comment (hoopoepg) [reply src/uct/sm/base/sm_ep.c]: it is possible via macro only, but in this case it bring complexity in debugging and error handling

File: src/uct/sm/self/self.c [hunk 3]
```diff
@@ -60,6 +60,19 @@ static ucs_status_t uct_self_iface_query(uct_iface_h tl_iface, uct_iface_attr_t
                                    UCT_IFACE_FLAG_CB_SYNC          |
                                    UCT_IFACE_FLAG_EP_CHECK;
 
+    attr->cap.atomic32.op_flags   =
```
[the remark this answers is not in the document]
Comment (hoopoepg) [reply src/uct/sm/self/self.c]: technically it is possible, but in this case all other stuff should be moved into separate call which is deeper refactoring (out of scope of this PR)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6698/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6699/ for details (Mellanox internal link).

**Pull request merged**